### ISSUE TITLE
curl: (43) A libcurl function was given a bad argument

### ISSUE THREAD

**Issue by ryandesign** [curl/curl#13958]
### I did this

```
curl -w '%{response_code}' https://github.com
```

### I expected the following

not a `43` error code

### curl/libcurl version

8.8.0

### operating system

macOS (issue seen on 12.7.5 and 14.5)

**Comment by bagder** [comment]
A full `curl -V` might help.

**Comment by ryandesign** [comment]
The bug report form is missing fields for most of the information I wanted to provide to you so I'll add it as a comment.

@ehjmx [reported to MacPorts](https://trac.macports.org/ticket/70226) that when MacPorts curl is built with http3 support (which uses gnutls) then `curl -Is -w '%header{location}' -o /dev/null https://github.com/SAP/macOS-icon-generator/releases/latest` results in an error 43 instead of the expected response. This looks like the same thing that was reported in #13845. I don't understand the resolution of that issue. @bagder suspected it was a "broken build" or a "build related issue"; I don't know what steps should be taken to correct those issues.

**Comment by ryandesign** [comment]
```
% curl -V
curl 8.8.0 (x86_64-apple-darwin21.6.0) libcurl/8.8.0 GnuTLS/3.7.11 zlib/1.3.1 brotli/1.1.0 zstd/1.5.6 libidn2/2.3.7 libpsl/0.21.5 nghttp2/1.62.1 ngtcp2/1.5.0 nghttp3/1.3.0
Release-Date: 2024-05-22
Protocols: dict file ftp ftps gopher gophers http https imap imaps ipfs ipns mqtt pop3 pop3s rtsp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS brotli HSTS HTTP2 HTTP3 HTTPS-proxy IDN IPv6 Largefile libz NTLM PSL SSL threadsafe TLS-SRP UnixSockets zstd
```

**Comment by bagder** [comment]
Super annoying error message :angry:  . I built a curl 8.8.0 gnutls+ngtcp2 version on my Linux box but it did not reproduce.

Any chance someone who can reproduce can step through the `-w` code to see which libcurl function that returns this? (and then I can work on improving the message for the future)

**Comment by ryandesign** [comment]
I should have read through #13845 more carefully. 9aa1d412b814a40868558da51a6ab28ce1384a58 does fix this problem.

**Comment by bagder** [comment]
Ugha, thanks. I got it. So fixed already via #13858.

Also, mitigated by #13914 (0a3c78f6b657f1e) but probably not terribly relevant now.

**Comment by bagder** [comment]
(But still curious why I did not see the problem.)

**Comment by vszakats** [comment]
> The bug report form is missing fields for most of the information I wanted to provide to you so I'll add it as a comment.
> 
> @ehjmx [reported to MacPorts](https://trac.macports.org/ticket/70226) that when MacPorts curl is built with http3 support (which uses gnutls) then `curl -Is -w '%header{location}' -o /dev/null https://github.com/SAP/macOS-icon-generator/releases/latest` results in an error 43 instead of the expected response. This looks like the same thing that was reported in #13845. I don't understand the resolution of that issue. @bagder suspected it was a "broken build" or a "build related issue"; I don't know what steps should be taken to correct those issues.

The workaround for #13845 was to avoid SecureTransport, and the fixes merged were: #13857 #13858.
Those were made to x509asn1, which is also used by GnuTLS, Schannel and wolfSSL, so these fixes might apply here.

**Comment by bagder** [comment]
The conclusion is that this regression is fixed already.
